### PR TITLE
Add guidelines on contributing code

### PR DESCRIPTION
**Pull request by talha131**

### CODE REVIEW

**Comment by iranzo** [comment]
Use Gerrit instead of GitHub PRs...

**Comment by iranzo** [comment]
IMHO, general github workflow is to fork, create new branch and submit pr, this still leaves lot of 'orphan' repos.

Gerrit instead does just keep one repo for commits and each user does submit reviews against it via 'gerrithub'

There are several reviews about them with analysis like <https://www.beepsend.com/2016/04/05/abandoning-gitflow-github-favour-gerrit/> 

In my case, gerrit seems easy as there's no need to pull changes from upstream repo, merge in my fork, etc, and getting a 'PR' (review) downloaded and tested is easy as (git-review -d $reviewid)

Each review gets commends, feedback on lines and +2 for code, +1 on verified (UT's) and then, can be merged, no leftover branches on master, all user can, when sending a new PR, use latest master, etc

**Comment by talha131** [comment]
Also, once 2.0 is release, we should do all dev against `next` branch. And accept PRs against it. It will be understood `next` refers to the very next milestone planned.

Or we can start following this practice immediately. Let me know what do you guys think?

**Comment by iranzo** [comment]
Doing on 'next' sounds good, but we do also need to backport fixes to master that break functionality (http/https, and others) as part of the regular process

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -45,6 +45,12 @@ Specifically:
 
 Refer to [Google's HTML/CSS Style Guide][google_style_guide] for all other formatting rules.
 
+# Contributing Code
+
+1. Create a new git branch specific to your change instead of to making your commits in the master branch.
+1. Create pull reuqest from that branch.
```
Comment (AWegnerGitHub) [inline CONTRIBUTING.md]: Tiny typo on this line. Should be "request"

Comment (talha131) [reply CONTRIBUTING.md]: Thank you @AWegnerGitHub

**Comment by talha131** [comment]
Guys I have updated the docs. If you can please review and approve it, I would be able to squash and merge this PR.

**Comment by talha131** [comment]
Thanks everyone for your input.

**Comment by iranzo** [comment]
@silverhook do you merge it then?

**Comment by iranzo** [comment]
It happened in parallel, thanks!

**Comment by talha131** [comment]
@iranzo I merged it just now 😆

---

yeah parallel!

**Pull request merged**